### PR TITLE
WIP: Ports RBMK Reactor from NSV13

### PR DESCRIPTION
**Pull request by silverplatedelta**
## What Does This PR Do
WORK IN PROGRESS

This PR ports the RMBK fission reactor from NSV13. This PR will not make the item and systems available to the playerbase. This is due to mapping requirements needed for the reactor to function as intended. PR opened for ease of assistance requests on a couple issues.

TODO:
- [ ] get reactor to turn on
- [ ] ensure proper node function 
- [ ] fix moderator input issues
- [ ] fix meltdown alarm system
- [ ] fix reactor ambience and sound
- [x] Setup computer console's ability to be constructed and deconstructed
- [ ] TGUI cleanup
- [ ] Balance reactor

## Why It's Good For The Game
Engineering content is always exciting. This port, while not player facing, will give developers (burzah and I) time to refine the systems before player facing implantation.

## Images of changes
TBA

## Testing
TBA

## Changelog
:cl: Silverplate, Burzah, and Programs-The-Station
add: Ports RBMK reactor from NSV13
/:cl:

### CODE REVIEW

**Comment by silverplatedelta** [comment]
Please do not add this PR to the mapping que. the mapping changes are to prevent me from having to scratchbuild a reactor any time I want to test it

File: code/modules/power/rbmk/rbmk.dm [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+
+	// below vars contol emergency alert systems
+	var/obj/item/radio/radio
+	var/rbmk_warning_delay = 60
+	var/lastrbmkwarn = 0
+	var/rbmk_safe_alert = "Reactor core returning to safe operating parameters"
+	var/rbmk_temp_alert = "Reactor core internal temperature rising above critical levels!"
+	var/rbmk_pressure_alert = "Reactor core internal pressure rising above critical levels!"
+	var/rbmk_meltdown_alert = "REACTOR MELTDOWN IMMINENT"
+	var/rbmk_blowout_alert = "REACTOR BLOWOUT IMMINENT"
```
Comment (Sirryan2002) [inline code/modules/power/rbmk/rbmk.dm:132]: make the comment documentation use proper autodoc format

File: code/game/machinery/computer/rbmkcontrol.dm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		ui.set_autoupdate(TRUE)
+
+/obj/machinery/computer/reactor/control_rods/ui_act(action, params)
+	if(..())
+		return
+	if(!reactor)
+		return
+	if(action == "input")
+		var/input = text2num(params["target"])
+		reactor.desired_k = CLAMP(input, 0, 3)
```
[the remark this answers is not in the document]
Comment (silverplatedelta) [reply code/game/machinery/computer/rbmkcontrol.dm]: define was removed, I was unaware the proc existed.

File: code/game/area/areas.dm [hunk 3]
```diff
@@ -78,6 +78,11 @@
 
 	var/area/area_limited_icon_smoothing
 
+	///used in RBMK and Stormdrive engines for ambience
+	var/ambient_buzz = 'sound/ambience/shipambience.ogg'
```
[the remark this answers is not in the document]
Comment (silverplatedelta) [reply code/game/area/areas.dm:82]: I'm unsure if I'm going to use this ambience, as ours is currently handled by area and is fine. I will address this review if/when that choice is made.

**Review by S34NW — COMMENTED** [review]
No commits in 2 months, still in draft, it might be better if this PR is closed for now and re-opened once you have made further progress and/or have more time for it?

**Comment by silverplatedelta** [comment]
Per Sean/dreamer suggestion, I'm going to be closing this PR until I have more time and the requisite skill to progress this project. Thank you to everyone who helped me get this far, I do hope we can finish this in the future.

**Pull request closed without merging**